File: ManualTransition/Week2 Selenium Rebuild Properly/Practical/selenium-page-object-model-practice/src/main/java/com/techlambda/selenium/pom/pages/CartPage.java
package com.techlambda.selenium.pom.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * CartPage owns cart-related locators and reusable actions.
 */
public class CartPage {

    private final WebDriver driver;

    private final By cartCount = By.id("cart-count");
    private final By cartItems = By.cssSelector("#cart-items li");
    private final By clearCartButton = By.id("clear-cart-button");

    public CartPage(WebDriver driver) {
        this.driver = driver;
    }

    public String getCartCountText() {
        return driver.findElement(cartCount).getText();
    }

    public int getCartItemCount() {
        return driver.findElements(cartItems).size();
    }

    public void clearCart() {
        driver.findElement(clearCartButton).click();
    }
}

File: PlayWriter/Practical/Advance/playwright-pom-practice/pages/CartPage.ts
import { Locator, Page } from '@playwright/test';

export class CartPage {
  readonly page: Page;
  readonly section: Locator;
  readonly cartItems: Locator;
  readonly emptyCartMessage: Locator;
  readonly cartCount: Locator;
  readonly totalItems: Locator;
  readonly clearCartButton: Locator;

  constructor(page: Page) {
    this.page = page;
    this.section = page.locator('#cart-section');
    this.cartItems = page.locator('#cart-items li');
    this.emptyCartMessage = page.locator('#empty-cart-message');
    this.cartCount = page.getByTestId('cart-count');
    this.totalItems = page.getByTestId('total-items');
    this.clearCartButton = page.getByRole('button', { name: 'Clear Cart' });
  }

  async clearCart() {
    await this.clearCartButton.click();
  }
}
